agenta-web/src/lib/helpers/observability_helpers.ts

```diff
@@ -0,0 +1,76 @@
+import {_AgentaRootsResponse, AgentaNodeDTO, AgentaTreeDTO} from "@/services/observability/types"
+
+export const observabilityTransformer = (
+    item: AgentaTreeDTO | AgentaNodeDTO,
+): _AgentaRootsResponse[] => {
+    const buildData = (node: AgentaNodeDTO) => {
+        const key = node.node.id
+        const hasChildren = node.nodes && Object.keys(node.nodes).length > 0
+
+        return {
+            ...node,
+            key,
+            ...(hasChildren ? {children: observabilityTransformer(node)} : undefined),
+        }
+    }
+
+    if (item.nodes) {
+        return Object.entries(item.nodes)
+            .flatMap(([_, value]) => {
+                if (Array.isArray(value)) {
+                    return value.map((item, index) =>
+                        buildData({
+                            ...item,
+                            node: {...item.node, name: `${item.node.name}[${index}]`},
+                        }),
+                    )
+                } else {
+                    return buildData(value)
+                }
+            })
+            .filter((node): node is _AgentaRootsResponse => node !== null && node !== undefined)
+    }
+
+    return []
+}
+
+export const buildNodeTree = ({parent, ...node}: AgentaNodeDTO) => ({
+    tree: node.tree.id,
+    nodes: [{...node}],
+})
+
+export const getNodeById = (
```

Comment:
Is this still relevant/correct with the new response format where nodes is always an array ? -- @bekossy